src/lib/operators/join_sort_merge/column_materializer.hpp

```diff
@@ -139,104 +122,30 @@ class ColumnMaterializer {
     }
   }
 
-  /**
-   * Materialization works of all types of segments
-   */
-  std::shared_ptr<MaterializedSegment<T>> _materialize_generic_segment(const AbstractSegment& segment,
-                                                                       const ChunkID chunk_id,
-                                                                       std::unique_ptr<RowIDPosList>& null_rows_output,
-                                                                       Subsample<T>& subsample) {
+  MaterializedSegment<T> _materialize_segment(const std::shared_ptr<AbstractSegment>& segment, const ChunkID chunk_id,
+                                              RowIDPosList& null_rows_output, Subsample<T>& subsample) {
     auto output = MaterializedSegment<T>{};
-    output.reserve(segment.size());
+    output.reserve(segment->size());
 
-    segment_iterate<T>(segment, [&](const auto& position) {
+    segment_iterate<T>(*segment, [&](const auto& position) {
       const auto row_id = RowID{chunk_id, position.chunk_offset()};
       if (position.is_null()) {
         if (_materialize_null) {
-          null_rows_output->emplace_back(row_id);
+          null_rows_output.emplace_back(row_id);
         }
       } else {
         output.emplace_back(row_id, position.value());
       }
     });
 
     if (_sort) {
-      std::sort(output.begin(), output.end(),
-                [](const auto& left, const auto& right) { return left.value < right.value; });
-    }
-
-    _gather_samples_from_segment(output, subsample);
-
-    return std::make_shared<MaterializedSegment<T>>(std::move(output));
-  }
-
```

Comment:
Why was this whole block removed?

Reply:
The dictionary specialization turned out to be a little bit slower that `std::sort` and now with `boost::sort` it was clearly slower that the generalized path.